server/sonar-server/src/test/java/org/sonar/server/computation/step/PersistDuplicationMeasuresStepTest.java

```diff
@@ -0,0 +1,393 @@
+    return step;
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    session.close();
+  }
+
+  @Test
+  public void nothing_to_do_when_no_duplication() throws Exception {
+    saveDuplicationMetric();
+
+    File reportDir = temp.newFolder();
+    BatchReportWriter writer = new BatchReportWriter(reportDir);
+    writer.writeMetadata(BatchReport.Metadata.newBuilder()
+      .setRootComponentRef(1)
+      .setProjectKey("PROJECT_KEY")
+      .setAnalysisDate(150000000L)
+      .build());
+
+    writer.writeComponent(BatchReport.Component.newBuilder()
+      .setRef(1)
+      .setType(Constants.ComponentType.PROJECT)
+      .setUuid("UUID_A")
+      .setKey("PROJECT_KEY")
+      .setSnapshotId(10L)
+      .addChildRef(2)
+      .build());
+    writer.writeComponent(BatchReport.Component.newBuilder()
+      .setRef(2)
+      .setType(Constants.ComponentType.FILE)
+      .setUuid("UUID_B")
+      .setSnapshotId(11L)
+      .setPath("file")
+      .build());
+
+    step.execute(new ComputationContext(new BatchReportReader(reportDir), ComponentTesting.newProjectDto("PROJECT")));
+
+    assertThat(dbTester.countRowsOfTable("project_measures")).isEqualTo(0);
+  }
+
+  @Test
+  public void persist_duplications_on_same_file() throws Exception {
+    MetricDto duplicationMetric = saveDuplicationMetric();
+
+    File reportDir = temp.newFolder();
+    BatchReportWriter writer = new BatchReportWriter(reportDir);
+    writer.writeMetadata(BatchReport.Metadata.newBuilder()
+      .setRootComponentRef(1)
+      .setProjectKey("PROJECT_KEY")
+      .setAnalysisDate(150000000L)
+      .build());
+
+    writer.writeComponent(BatchReport.Component.newBuilder()
+      .setRef(1)
+      .setType(Constants.ComponentType.PROJECT)
+      .setUuid("UUID_A")
+      .setKey("PROJECT_KEY")
+      .setSnapshotId(10L)
+      .addChildRef(2)
+      .build());
+    writer.writeComponent(BatchReport.Component.newBuilder()
+      .setRef(2)
+      .setType(Constants.ComponentType.FILE)
+      .setUuid("UUID_B")
+      .setSnapshotId(11L)
+      .setPath("file")
+      .build());
+
+    BatchReport.Duplication duplication = BatchReport.Duplication.newBuilder()
+      .setOriginBlock(BatchReport.DuplicationBlock.newBuilder()
+        .setOtherComponentRef(2)
+        .setStartLine(1)
+        .setEndLine(5)
+        .build())
+      .addDuplicatedBy(BatchReport.DuplicationBlock.newBuilder()
+        .setOtherComponentRef(2)
+        .setStartLine(6)
+        .setEndLine(10)
+        .build())
+      .build();
+    writer.writeComponentDuplications(2, newArrayList(duplication));
+
+    step.execute(new ComputationContext(new BatchReportReader(reportDir), ComponentTesting.newProjectDto("PROJECT")));
+
+    assertThat(dbTester.countRowsOfTable("project_measures")).isEqualTo(1);
+
+    Map<String, Object> dto =  dbTester.selectFirst("select snapshot_id as \"snapshotId\", metric_id as \"metricId\", text_value as \"textValue\" from project_measures");
```

Comment:
Removing them make the test fail, because without quote, and for an unknown reason, column name will be returned in upper case.
But we should improve our dbTester framework to simplify such query. No idea for the moment on how to do that.